### ISSUE TITLE
Withdrawals and transfers should be reflected in the vault

### ISSUE THREAD

**Issue by stann1** [podkrepi-bg/api#272]
Scope:

1. Add the inter-dependencies between withdrawals, transfers, donations and vaults
2. Implement ACID transactions for each vault activity.

The following changes are being implemented for vaults:
- new field - blocked amount
- the actual available amount used for further calculations and for user viewing, should always be the difference between amount and blocked amount
- entity deletion is no longer possible (in a separate issue deletion will be changed to soft-delete, i.e. flag the entity)

The following changes are being implemented for withdrawals,transfers,expenses:
- entity creation will now create the entity and block the amount in the corresponding source vault, using the 'blockedAmount' field (e.g. the value of the field will increase with X)
- entity update will not be possible if the source/target vault or amount is changed, an error will be thrown
- entity update where the approval status is changed from **pending** to **completed** will unblock the previously blocked amount, and will debit(subtract) it from the source vault's balance. In case of a transfer, the target vault's balance will be credited
- entity update where no status or vault is being changed will simply update the entity fields
- entity deletion is no longer possible (in a separate issue deletion will be changed to soft-delete, i.e. flag the entity)
